lib/user_app_jwt/user_app_jwt.c

```diff
@@ -0,0 +1,758 @@
+		// UUID byte endiannes is little endian first.
+		sprintf(temp_str[i], "%08x", swap_endian(uuid_words[i]));
+	}
+
+	// UUID string format defined by  ITU-T X.667 | ISO/IEC 9834-8 :
+	// <8 char>-<4 char>-<4 char>-<4 char>-<12 char>
+
+	// for some reason, sprintf will not respect the amout of characters to print that is
+	// specified in the formatting string. use this cascade of sprintf to overcome this issue.
+	// 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0
+	//'^         ' = "59fa2510"
+	sprintf(uuid_str_out, "%8s", temp_str[0]);
+	// 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0
+	//			   '^   ' = "8952"
+	sprintf(uuid_str_out + 8, "-%4s", temp_str[1] + 0);
+	// 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0
+	//			       '^   ' = "abc5"
+	sprintf(uuid_str_out + 8 + 5, "-%4s", temp_str[1] + 4);
+	// 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0
+	//  			            '^   ' = "a20c"
+	sprintf(uuid_str_out + 8 + 5 + 5, "-%4s", temp_str[2] + 0);
+	// 59fa2510\0 , 8952abc5\0 , a20cd981\0 , 2bf2e936\0
+	//  			                '^   ' + '^         '  = "d9812bf2e936"
+	sprintf(uuid_str_out + 8 + 5 + 5 + 5, "-%4s%8s", temp_str[2] + 4, temp_str[3]);
+	uuid_str_out[NRF_UUID_V4_STR_LEN] = '\0';
+
+	return 0;
+}
+
+#if defined(CONFIG_PRINT_EXPORTED_PUBKEY_DER)
+//---------------------------------------------------------------------------//
+static int bytes_to_str(const uint8_t *const bytes_buf, const size_t bytes_len, char *string_buff,
+			const size_t string_buff_size, size_t *string_len)
+{
+	int err = 0;
+
+	if ((NULL != bytes_buf) && (0 != bytes_len) && (NULL != string_buff) &&
+	    (0 != string_buff_size)) {
+		uint32_t str_pos = 0;
+		uint8_t byte_h, byte_l = 0;
+
+		for (uint32_t count = 0; count < bytes_len; count++) {
+			byte_h = ((bytes_buf[count] >> 4) & 0xF);
+			byte_l = (bytes_buf[count] & 0x0F);
+			if (byte_h < 10) {
+				/* transpose from 0-9 integer numbers */
+				/* to 0x30('0') - 0x31('9') ASCII characters */
+				byte_h += '0';
+			} else {
+				/* transpose from 10-15 integer numbers */
+				/* to 0x61('a') - 0x66('f') ASCII characters */
+				byte_h += ('a' - 10);
+			}
+
+			if (byte_l < 10) {
+				byte_l += '0';
+			} else {
+				byte_l += ('a' - 10);
+			}
+
+			sprintf(string_buff + str_pos, "%c%c", (char)(byte_h), (char)(byte_l));
+			str_pos += 2;
+		}
+
+		string_buff[str_pos] = '\0';
+		*string_len = str_pos + 1;
+	} else {
+		LOG_ERR("bytes_to_str : bad parameters");
+		err = -1;
+	}
+
+	return err;
+}
+#endif /* CONFIG_PRINT_EXPORTED_PUBKEY_DER */
+
+#if !defined(CONFIG_DATE_TIME)
+//---------------------------------------------------------------------------//
+int date_time_now(uint64_t *timestampe)
+{
+	// For test purposes, use a simulated value of time
+	// Wednesday, December 4, 2024 3:00:43 PM GMT+02:00
+
+	*timestampe = 1733317243000;
+	return 0;
+}
+#endif
+
+//---------------------------------------------------------------------------//
+static int crypto_init(void)
```

Comment:
This function is called from several places. I assume, that the initialization has to be only done once? If yes, then you could do the initialization with `SYS_INIT()` instead, for example: https://github.com/nrfconnect/sdk-nrf/blob/b0461ac723c3e66f35c829a62223f9cbc2ea34d0/lib/lte_link_control/lte_lc.c#L241